src/layout/appHeader.tsx

```diff
@@ -1,34 +1,159 @@
-import React, {FC} from 'react'
-import {Link} from 'react-router-dom'
-import {AppBar, Toolbar, Typography, Button} from '@material-ui/core'
+import React, {FC, Dispatch, SetStateAction} from 'react'
+import {useHistory} from 'react-router-dom'
+import {
+  AppBar,
+  Toolbar,
+  Typography,
+  Button,
+  makeStyles,
+  IconButton,
+  Menu,
+  MenuItem,
+  useMediaQuery,
+  useTheme,
+  Divider,
+  ListSubheader,
+} from '@material-ui/core'
+import {AccountCircle} from '@material-ui/icons'
+import MenuIcon from '@material-ui/icons/Menu'
 import {useAuth0} from '@auth0/auth0-react'
+import {routes} from '../routes'
 
-export const AppHeader: FC = props => {
-  const {isAuthenticated, loginWithPopup, logout} = useAuth0()
+interface Props {
+  sideDrawOpen: boolean
+  setSideDrawerOpen: Dispatch<SetStateAction<boolean>>
+}
+
+/**
+ * Opening the user menu and side drawer generates a warning in the console
+ * Apparently this is a known issue when running material ui in strict mode https://github.com/mui-org/material-ui/issues/13394
+ * From my reading this will not happen in production and can be resolved by removing strict mode
+ */
+
+export const AppHeader: FC<Props> = ({sideDrawOpen, setSideDrawerOpen}) => {
+  const {isAuthenticated, loginWithPopup, logout, user} = useAuth0()
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null)
+  const userMenuOpen = Boolean(anchorEl)
+  const theme = useTheme()
+  const history = useHistory()
+  const isMobile = useMediaQuery(theme.breakpoints.down('xs'))
+
+  const useStyles = makeStyles(theme => ({
+    root: {
+      flexGrow: 1,
+    },
+    loginButton: {
+      marginLeft: theme.spacing(1),
+    },
+    menuButton: {
+      marginLeft: theme.spacing(1),
+    },
+    title: {
+      flexGrow: 1,
+    },
+    navLink: {
+      padding: theme.spacing(1),
+    },
+  }))
```

Comment:
What do you guys think if we move this separate file in case we see it grow?

Reply:
My feeling is that these styles are very specific to the app header and probably won't be used elsewhere within the code base. I am sure that we will use a separate styles file as the project grows and I would be happy to move these to there if required. I would need to make the class names a lot more specific though.